internal/tss/services/transaction_service_test.go

```diff
@@ -79,7 +79,204 @@ func TestBuildPayments(t *testing.T) {
+		channelAccountSignatureClient.
+			On("GetAccountPublicKey", context.Background()).
+			Return(channelAccount.Address(), nil).
+			Once()
+
+		horizonClient.
+			On("AccountDetail", horizonclient.AccountRequest{
+				AccountID: channelAccount.Address(),
+			}).
+			Return(horizon.Account{AccountID: channelAccount.Address(), Sequence: 1}, nil).
+			Once()
+
+		tx, err := txService.BuildAndSignTransactionWithChannelAccount(context.Background(), []txnbuild.Operation{}, 30)
+		assert.Empty(t, tx)
+		assert.Equal(t, "building transaction: transaction has no operations", err.Error())
+
+	})
+
+	t.Run("sign_stellar_transaction_w_channel_account_err", func(t *testing.T) {
+		channelAccount := keypair.MustRandom()
+		channelAccountSignatureClient.
+			On("GetAccountPublicKey", context.Background()).
+			Return(channelAccount.Address(), nil).
+			Once().
+			On("SignStellarTransaction", context.Background(), mock.AnythingOfType("*txnbuild.Transaction"), []string{channelAccount.Address()}).
+			Return(nil, errors.New("unable to sign")).
+			Once()
+
+		horizonClient.
+			On("AccountDetail", horizonclient.AccountRequest{
+				AccountID: channelAccount.Address(),
+			}).
+			Return(horizon.Account{AccountID: channelAccount.Address(), Sequence: 1}, nil).
+			Once()
+
+		payment := txnbuild.Payment{
+			Destination:   keypair.MustRandom().Address(),
+			Amount:        "10",
+			Asset:         txnbuild.NativeAsset{},
+			SourceAccount: keypair.MustRandom().Address(),
+		}
+		tx, err := txService.BuildAndSignTransactionWithChannelAccount(context.Background(), []txnbuild.Operation{&payment}, 30)
+		assert.Empty(t, tx)
+		assert.Equal(t, "signing transaction with channel account: unable to sign", err.Error())
+	})
+
+	t.Run("returns_signed_tx", func(t *testing.T) {
+		signedTx := utils.BuildTestTransaction()
+		channelAccount := keypair.MustRandom()
+		channelAccountSignatureClient.
+			On("GetAccountPublicKey", context.Background()).
+			Return(channelAccount.Address(), nil).
+			Once().
+			On("SignStellarTransaction", context.Background(), mock.AnythingOfType("*txnbuild.Transaction"), []string{channelAccount.Address()}).
+			Return(signedTx, nil).
+			Once()
+
+		horizonClient.
+			On("AccountDetail", horizonclient.AccountRequest{
+				AccountID: channelAccount.Address(),
+			}).
+			Return(horizon.Account{AccountID: channelAccount.Address(), Sequence: 1}, nil).
+			Once()
+
+		payment := txnbuild.Payment{
+			Destination:   keypair.MustRandom().Address(),
+			Amount:        "10",
+			Asset:         txnbuild.NativeAsset{},
+			SourceAccount: keypair.MustRandom().Address(),
+		}
+		tx, err := txService.BuildAndSignTransactionWithChannelAccount(context.Background(), []txnbuild.Operation{&payment}, 30)
+		assert.Equal(t, signedTx, tx)
+		assert.NoError(t, err)
+	})
+}
+
+func TestBuildFeeBumpTransaction(t *testing.T) {
+	distributionAccountSignatureClient := signing.SignatureClientMock{}
+	defer distributionAccountSignatureClient.AssertExpectations(t)
```

Comment:
Hmm, shouldnt this be in each separate test case instead of a defer statement? Since we are setting different expectations in each case for this mock client, we would need to assert this in each test run

Reply:
I thought that if I called distributionAccountSignatureClient.AssertExpectations(t) with a defer, it would effectively call this function for ALL the individual test cases at the end of this test case. But I changed it so that I'm calling it in each individual function